### PR TITLE
switch default ciphersuite to 2

### PR DESCRIPTION
**Pull request by stefanwire**
https://wearezeta.atlassian.net/browse/WPB-15004

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: integration/test/Test/MLS/KeyPackage.hs [hunk 1]
```diff
@@ -11,9 +11,10 @@ import Testlib.Prelude
 
 testDeleteKeyPackages :: App ()
 testDeleteKeyPackages = do
+  let suite = Ciphersuite "0x0001"
   alice <- randomUser OwnDomain def
-  alice1 <- createMLSClient def def alice
-  kps <- replicateM 3 (uploadNewKeyPackage def alice1)
+  alice1 <- createMLSClient suite def alice
+  kps <- replicateM 3 (uploadNewKeyPackage suite alice1)
```
Comment (akshaymankar) [inline integration/test/Test/MLS/KeyPackage.hs:17]: Why this change? I thought we wanted most tests to use `def` suite?

**Review by akshaymankar — COMMENTED** [review]
We should add release notes about this change because if the operator doesn't notice this, the clients could get into a very confusing.

File: libs/wire-api/src/Wire/API/Conversation/Protocol.hs [hunk 2]
```diff
@@ -109,7 +109,7 @@ optionalActiveMLSConversationDataSchema (Just v)
             schema
         <*> fmap (.epochTimestamp)
           .= field "epoch_timestamp" (named "EpochTimestamp" . nullable . unnamed $ utcTimeSchema)
-        <*> maybe MLS_128_DHKEMX25519_AES128GCM_SHA256_Ed25519 (.ciphersuite)
+        <*> maybe MLS_128_DHKEMP256_AES128GCM_SHA256_P256 (.ciphersuite)
```
Comment (akshaymankar) [inline libs/wire-api/src/Wire/API/Conversation/Protocol.hs:112]: This looks like a mistake, we shouldn't change legacy defaults in older versions.

File: libs/wire-api/src/Wire/API/MLS/CipherSuite.hs [hunk 3]
```diff
@@ -122,7 +123,13 @@ data CipherSuiteTag
   deriving (Arbitrary) via (GenericUniform CipherSuiteTag)
 
 defCipherSuite :: CipherSuiteTag
-defCipherSuite = MLS_128_DHKEMX25519_AES128GCM_SHA256_Ed25519
+defCipherSuite = MLS_128_DHKEMP256_AES128GCM_SHA256_P256
```
Comment (akshaymankar) [inline libs/wire-api/src/Wire/API/MLS/CipherSuite.hs:126]: This is unused now. As discussed on the call, we should delete this and make a v8 version of the only endpoint which uses `defaultCipherSuiteV7` which makes the ciphersuite mandatory.

Comment (akshaymankar) [reply libs/wire-api/src/Wire/API/MLS/CipherSuite.hs:126]: Also feel free to do this in a separate PR/ticket.

Comment (stefanwire) [reply libs/wire-api/src/Wire/API/MLS/CipherSuite.hs]: Now done in this PR.

**Review by akshaymankar — APPROVED** [review]
Looks good :rocket:
